### PR TITLE
LPS-169584

### PR DESCRIPTION
**Pull request by gabrielwas**
Related issues:

https://issues.liferay.com/browse/LPS-169582
https://issues.liferay.com/browse/LPS-169584
https://issues.liferay.com/browse/LPS-169878
https://issues.liferay.com/browse/LPS-169879

@danvb112 was part of the FE code

### CODE REVIEW

**Comment by gabrielwas** [comment]
cc @danvb112

**Comment by gabrielwas** [comment]
ci:test:relevant

**Comment by timpak** [comment]
ci:test:object

**Comment by timpak** [comment]
ci:test:bundle

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java [hunk 1]
```diff
@@ -2189,6 +2190,19 @@ else if (Objects.equals(
...
+							objectFilter.getFilterType(),
+							ObjectFilterConstants.TYPE_CURRENT_USER)) {
+
+						objectFilter.setJSON(
+							String.valueOf(
+								_jsonFactory.createJSONObject(
+								).put(
+									"currentUserId",
+									PrincipalThreadLocal.getUserId()
+								)));
```
Comment (carolmariaabb) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java:2203]: @gabrielwas can we change to:

```
objectFilter.setJSON(
	JSONUtil.put(
		"currentUserId", PrincipalThreadLocal.getUserId()
	).toString()
)
```
or do you prefer to use the `_jsonFactory`?

Comment (gabrielwas) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java:2203]: @carolmariaabb better your way! Thanks!

**Comment by gabrielwas** [comment]
Hey @timpak, I'm fixing [com.liferay.object.service.test.ObjectActionLocalServiceTest](https://test-1-33.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/183947/testReport/com.liferay.object.service.test/ObjectActionLocalServiceTest)

**Comment by gabrielwas** [comment]
ci:reevaluate:1257401_5563

**Comment by gabrielwas** [comment]
ci:test:sf

**Comment by gabrielwas** [comment]
ci:test:relevant

**Comment by timpak** [comment]
Hey @gabrielwas, looks like you need to run baseline. Besides that, I didn't see any test failures that were related to your changes. I'm going to manually test the use case described in [LPS-169575](https://issues.liferay.com/browse/LPS-169575).

>      
     [exec]   PACKAGE_NAME                                       DELTA      CUR_VER    BASE_VER   REC_VER    WARNINGS  
     [exec] = ================================================== ========== ========== ========== ========== ==========
     [exec] * com.liferay.object.constants                       MINOR      5.1.0      5.1.0      5.2.0      VERSION INCREASE REQUIRED
     [exec] 	<   class      com.liferay.object.constants.ObjectFilterConstants
     [exec] 		+   field      TYPE_CURRENT_USER
     [exec] 			+   access     final
     [exec] 			+   access     static
     [exec] 			+   constant   currentUser
     [exec] 			+   return     java.lang.String
     [exec] * com.liferay.object.exception                       MINOR      13.0.0     13.0.0     13.1.0     VERSION INCREASE REQUIRED
     [exec] 	+   class      com.liferay.object.exception.ObjectEntryValuesException$InvalidObjectStateTransition

**Comment by timpak** [comment]
ci:test:bundle

**Comment by gabrielwas** [comment]
Sent to Brian here https://github.com/brianchandotcom/liferay-portal/pull/127029

**Pull request closed without merging**